File: concrete/src/test/java/pcb/uwu/amount/derived/thermodynamics/PascalsTest.java
package pcb.uwu.amount.derived.thermodynamics;

import org.junit.Test;
import pcb.uwu.amount.derived.fundamental.Area;
import pcb.uwu.amount.derived.fundamental.Volume;
import pcb.uwu.amount.derived.mechanics.Newtons;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.derived.area.SquareMeterUnit.SQUARE_METER;
import static pcb.uwu.unit.derived.fundamental.CubicMeterUnit.CUBIC_METER;

public class PascalsTest {
	private static final Pascals pascals = new Pascals(9);

	@Test
	public void multipliedByVolume() {
		assertEquals(new Joules(27),
				pascals.multipliedBy(new Volume(3, CUBIC_METER), DECIMAL64));
	}

	@Test
	public void multipliedByArea() {
		assertEquals(new Newtons(18),
				pascals.multipliedBy(new Area(2, SQUARE_METER), DECIMAL64));
	}
}

File: concrete/src/test/java/pcb/uwu/amount/derived/thermodynamics/PascalsTest.kt
package pcb.uwu.amount.derived.thermodynamics

import org.junit.Assert.assertEquals
import org.junit.Test
import pcb.uwu.amount.derived.fundamental.Area
import pcb.uwu.amount.derived.fundamental.Volume
import pcb.uwu.unit.derived.area.SquareMeterUnit
import pcb.uwu.unit.derived.fundamental.CubicMeterUnit.CUBIC_METER
import java.math.MathContext.DECIMAL64

class PascalsTest
{
    @Test
    fun multipliedByVolume() =
        assertEquals("27 J",
                     Pascals(9).multipliedBy(Volume(amount = 3, unit = CUBIC_METER), DECIMAL64).toString())

    @Test
    fun multipliedByArea() =
        assertEquals("18 N",
                     Pascals(9).multipliedBy(Area(amount = 2, unit = SquareMeterUnit.SQUARE_METER), DECIMAL64).toString())

}